// firestore-export-import/dist/helper.js
import { GeoPoint as i } from "firebase-admin/firestore";
const u = (n) => !n._latitude || !n._longitude ? null : new i(n._latitude, n._longitude), f = (n) => !n || !n._seconds ? null : new Date(n._seconds * 1e3), _ = (n) => n && typeof n.path == "string" ? n.path : n, p = (n) => (n == null ? void 0 : n.constructor) === Object, y = (n) => Array.isArray(n), o = (n, t) => {
  var e;
  for (const [c, r] of Object.entries(n)) {
    if (!p(r) && !y(r) && ((e = r == null ? void 0 : r.constructor) == null ? void 0 : e.name) !== "DocumentReference")
      continue;
    const s = t(r);
    if (s) {
      n[c] = s;
      continue;
    }
    o(n[c], t);
  }
}, P = (n) => {
  o(n, (t) => typeof t == "object" && t.hasOwnProperty("_seconds") && t.hasOwnProperty("_nanoseconds") ? f(t) : null);
};
function h(n) {
  o(n, (t) => typeof t == "object" && t.hasOwnProperty("_latitude") && t.hasOwnProperty("_longitude") ? u(t) : null);
}
export {
  _ as getPath,
  u as makeGeoPoint,
  f as makeTime,
  P as parseAndConvertDates,
  h as parseAndConvertGeos,
  o as traverseObjects
};
